src/back/spv/writer.rs

```diff
@@ -2345,6 +2322,15 @@ impl Writer {
         ir_module: &crate::Module,
         mod_info: &ModuleInfo,
     ) -> Result<(), Error> {
+        let has_storage_buffers = ir_module
+            .global_variables
+            .iter()
+            .any(|(_, var)| var.class == crate::StorageClass::Storage);
+        if self.physical_layout.version < 0x10300 && has_storage_buffers {
+            // enable the storage buffer class on < SPV-1.3
+            Instruction::extension("SPV_KHR_storage_buffer_storage_class")
+                .to_words(&mut self.logical_layout.extensions);
+        }
```

Comment:
For now this is okay, maybe if we later get more use cases of checking the SPV version, and adding extensions, we can do it in a more elegant way. 😄

Reply:
Possibly, yes. This PR doesn't actually change the number of checks, it just moves this check into a better place :)